docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,96 @@
+
+## Migrating OpenShift Service Mesh 2.6 Cluster wide to OpenShift Service Mesh 3
+
+### Prerequisites
+- you have completed all the steps in the [pre-migration checklist](../README.md#pre-migration-checklist)
+- you have read [OpenShift Service Mesh 2 vs. OpenShift Service Mesh 3](../../ossm2-vs-ossm3.md)
+- you have verified that your 2.6 `ServiceMeshControlPlane` is using `ClusterWide` mode
+- Red Hat OpenShift Service Mesh 3 operator is installed
+- `IstioCNI` is installed
+- `istioctl` is installed
+
+### Procedure
+In this example, we'll be using the [bookinfo demo](https://raw.githubusercontent.com/Maistra/istio/maistra-2.6/samples/bookinfo/platform/kube/bookinfo.yaml) but you can follow these same steps with your own workloads.
+
+TODO: should we add a note about `minReplicas` for workload deployments?
+TODO: include also ingress gateway deployment migration?
+
+#### Plan the migration
+There will be two cluster wide Istio control planes running during the migration process so it's necessary to plan the migration steps in advance in order to avoid possible conflicts between the two control planes.
+
+There is a few conditions which must be assured to achieve the seamless migration:
+- both control planes must share the same root certificate
+
+  This can be achieved simply by installing the 3.0 control plane to the same namespace as 2.6 control plane
+- both control planes have access to all migrated namespaces and namespaces running addons (e.g. otel collector)
+  1. verify there are no Network Policies blocking the traffic
+  1. `discoverySelectors`  defined in 3.0 `Istio` resource must match migrated namespaces. See [Scoping the service mesh with DiscoverySelectors](../../create-mesh/README.md)
+- only one control plane will try to inject a side car
+
+To achieve last condition, it's necessary to understand how the injection works. Please see [Installing the Sidecar](../../injection/README.md) for details.
+> [!CAUTION]
+> In case you decide to use a `default` name for 3.0 `Istio` resource with `InPlace` upgrade strategy, 3.0 control plane will try to inject side cars to all pods in namespaces with `istio-injection=enabled` label and all pods with `sidecar.istio.io/inject="true"` label after next restart of the workloads. Without using `maistra.io/ignore-namespace: "true"` label, it will result in a conflict with 2.6 control plane injector.
+
+In case you are using `istio-injection=enabled` label on your data plane namespaces and planning to keep it, you should first read revision tag documentation. TODO: add link. TODO: describe cases where the migration will work with `istio-injection=enabled` (default revision, default revision tag)
```

Comment:
I think we should steer people as far away as possible from any scenario where an injection conflict will occur. Either by telling users to **not** name their `Istio` resource `default` and/or by only using `istio.io/rev`. In any case, users will likely need to relabel all of their dataplane namespaces? Either by adding `maistra.io/ignore-namespace` or by removing `istio-injection=enabled` in favor of `istio.io/rev`. The instructions could be more prescriptive and tell the user "do this" and "don't do this".

I prefer we tell users not to use the `default` name. And we have them label all their namespaces at once with `maistra.io/ignore-namespace: "true"`. If they want to keep `istio-injection=enabled`, then they should create a `RevisionTag`.

Reply:
If we tell users not to use `default` name, it would make the migration guide simpler for sure. But I'm not sure if we can do that as it's a valid use case. e.g. 2.6 users with `istio-injection=enabled` used everywhere who want to do a simple migration "all at once" would just: